### PR TITLE
Add an `--upgrade` flag to `./hack/update-deps.sh`

### PR DESCRIPTION
**Pull request by mattmoor**
This flag upgrades the list of dependencies that we choose to track at HEAD.

cc @n3wscott @coryrc @chaodaiG @vdemeester

### CODE REVIEW

File: hack/update-deps.sh [hunk 1]
```diff
@@ -23,8 +23,26 @@ set -o pipefail
...
+FLOATING_DEPS=(
+  "knative.dev/test-infra"
+)
+
+# Parse flags to determine any we should pass to dep.
+DEP_FLAGS=""
+while [[ $# -ne 0 ]]; do
+  parameter=$1
+  case ${parameter} in
+    --upgrade) DEP_FLAGS="-update ${FLOATING_DEPS[@]}" ;;
```
Comment (n3wscott) [inline hack/update-deps.sh]: can you try this script with two or more floaters? I thought there should only be one `-update` flag?

Comment (n3wscott) [reply hack/update-deps.sh]: OH nevermind... I forgot we are passing in --upgrade, not the deps.

IGNORE ME

also...

:fire: bash

Comment (mattmoor) [reply hack/update-deps.sh]: This only yields one flag, and I did try with two.  I have a tweak to this incoming to fix the error where `""` is passed to `dep` without `--upgrade`, testing locally now.

**Pull request merged**